server/app/controllers/dev/seeding/CategoryTranslationFileParser.java

```diff
@@ -0,0 +1,149 @@
+  public static final ImmutableMap<String, String> LO_CATEGORY_TRANSLATIONS =
+      parseFileForCategoryTranslations(
+          CATEGORY_TRANSLATIONS_DIRECTORY
+              + "lo_program_categories"
+              + CATEGORY_TRANSLATIONS_FILE_EXTENSION);
+
+  /** Maps each English category tag to its translation in Somali. */
+  public static final ImmutableMap<String, String> SO_CATEGORY_TRANSLATIONS =
+      parseFileForCategoryTranslations(
+          CATEGORY_TRANSLATIONS_DIRECTORY
+              + "so_program_categories"
+              + CATEGORY_TRANSLATIONS_FILE_EXTENSION);
+
+  /** Maps each English category tag to its translation in Tagalog. */
+  public static final ImmutableMap<String, String> TL_CATEGORY_TRANSLATIONS =
+      parseFileForCategoryTranslations(
+          CATEGORY_TRANSLATIONS_DIRECTORY
+              + "tl_program_categories"
+              + CATEGORY_TRANSLATIONS_FILE_EXTENSION);
+
+  /** Maps each English category tag to its translation in Vietnamese. */
+  public static final ImmutableMap<String, String> VI_CATEGORY_TRANSLATIONS =
+      parseFileForCategoryTranslations(
+          CATEGORY_TRANSLATIONS_DIRECTORY
+              + "vi_program_categories"
+              + CATEGORY_TRANSLATIONS_FILE_EXTENSION);
+
+  /** Maps each English category tag to its translation in Chinese. */
+  public static final ImmutableMap<String, String> ZH_CATEGORY_TRANSLATIONS =
+      parseFileForCategoryTranslations(
+          CATEGORY_TRANSLATIONS_DIRECTORY
+              + "zh-TW_program_categories"
+              + CATEGORY_TRANSLATIONS_FILE_EXTENSION);
+
+  public static final ImmutableList<String> PROGRAM_CATEGORY_NAMES =
+      getCategoryNamesFromTranslationMap(AM_CATEGORY_TRANSLATIONS);
+
+  private static final Logger logger = LoggerFactory.getLogger(CategoryTranslationFileParser.class);
+
+  public static CategoryModel createCategoryModelFromTranslationsMap(String category) {
+    String key = "tag." + category.toLowerCase(Locale.ROOT);
+    return new CategoryModel(
+        ImmutableMap.of(
+            Lang.forCode("am").toLocale(),
+            AM_CATEGORY_TRANSLATIONS.get(key),
+            Lang.forCode("ko").toLocale(),
+            KO_CATEGORY_TRANSLATIONS.get(key),
+            Lang.forCode("so").toLocale(),
+            SO_CATEGORY_TRANSLATIONS.get(key),
+            Lang.forCode("lo").toLocale(),
+            LO_CATEGORY_TRANSLATIONS.get(key),
+            Lang.forCode("tl").toLocale(),
+            TL_CATEGORY_TRANSLATIONS.get(key),
+            Lang.forCode("vi").toLocale(),
+            VI_CATEGORY_TRANSLATIONS.get(key),
+            Lang.forCode("en-US").toLocale(),
+            EN_CATEGORY_TRANSLATIONS.get(key),
+            Lang.forCode("es-US").toLocale(),
+            ES_CATEGORY_TRANSLATIONS.get(key),
+            Lang.forCode("zh-TW").toLocale(),
+            ZH_CATEGORY_TRANSLATIONS.get(key)));
+  }
+
+  private static ImmutableMap<String, String> parseFileForCategoryTranslations(String fileName) {
```

Comment:
**suggestion**: Instead of parsing the file by hand, I think we should use the `Properties` java util, like we do it `MessagesTest`:
https://github.com/civiform/civiform/blob/39cdc4699c6d0346c1466f04e55979c57486c40b/server/test/conf/MessagesTest.java#L157-L172

This gives us back a map of keys-value pairs that will be easier to work with in the rest of this file, and automatically ignores comment lines.

Reply:
I also wonder if it's easier to sort these values into category-specific maps at this point, as we read in each file, instead of pivoting these lists from language to category later when we're retrieving the category.

Reply:
I've replaced the parsing with the Java `Properties` util.  Thanks for that suggestion!  

I haven't had a chance to refactor so that we're sorting into category-specific maps here, but will do that when I get back.

Reply:
I refactored so that we sort into category-specific maps rather than first sorting into language-specific maps.  This meant hard-coding the categories because I needed to create category map variables before parsing all the files at once, but I think it's okay to do that since our pre-defined categories are fixed for now and once we make them more flexible, we likely won't be using the database seed tasks to create them anymore.